### DISCUSSION TITLE
MineColonies Addon

### DISCUSSION THREAD

**Discussion by Fernyam** [ldtteam/minecolonies#10499] [Q&A] [answered]
hello, I am currently developing an addon for MineColonies 1.21.1, which adds many new features related to multiplayer and war between colonies, and I would like to ask when I will finish production of the addon (The date is not known, but about 1-2 months), and if I give you the source code of the mod, then you can continue to study with this mod - to improve, fix bugs, transfer to new versions?

**Comment by Thodor12** [comment]
I doubt we would maintain anything that's not originally made by us. We have our own plans for these sort of mechanics, so if you want it in the mod you're better off contacting Raycoms and discussing with him to get it directly implemented into the mod itself.

There's no point for there to be an officially maintained addon to the base mod

**Comment by Raycoms** [comment, marked as answer]
We encourage people to get in contact with the leads on discord to talk about these things and already develop them in the core, if they align with our vision instead of creating a huge addon that would be very messy to merge. It's usually easier to develop the main parts in minecolonies and where the vision doesn't align add datapack support for example.
